lib/container-info.js

```diff
@@ -1,20 +1,103 @@
-const getContainerInfo = require('container-info')
-
-/**
- * Function that uses `container-info`
- * package to extract container information
- * and apply additional naming rules
- *
- * Injectable getInfo for testing purposes
- *
- * https://github.com/elastic/apm/blob/main/specs/agents/metadata.md#containerkubernetes-metadata
- */
-module.exports = (getInfo = getContainerInfo) => {
-  const info = getInfo.sync()
-  if (!info) { return info }
-
-  if (info.podId) {
-    info.podId = info.podId.replace(/_/g, '-')
+'use strict'
+
+const fs = require('fs')
+
+const uuidSource = '[0-9a-f]{8}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{4}[-_][0-9a-f]{12}'
+const containerSource = '[0-9a-f]{64}'
+const taskSource = '[0-9a-f]{32}'
+const awsEcsSource = '[0-9a-f]{32}-[0-9]{10}'
+
+const lineReg = /^(\d+):([^:]*):(.+)$/
+const podReg = new RegExp(`pod(${uuidSource})(?:.slice)?$`)
+const containerReg = new RegExp(`(${uuidSource}|${containerSource})(?:.scope)?$`)
+const taskReg = new RegExp(`^/ecs/(${taskSource})/.*$`)
+
+let ecsMetadata
+resetEcsMetadata(process.env.ECS_CONTAINER_METADATA_FILE)
+
+function resetEcsMetadata (file) {
+  ecsMetadata = ecsMetadataSync(file)
+}
+
+function parseLine (line) {
+  const [id, groups, path] = (line.match(lineReg) || []).slice(1)
+  const data = { id, groups, path }
+  const parts = path.split('/')
+
+  const controllers = groups.split(',')
+  if (controllers) data.controllers = controllers
+
+  const containerId = (parts.pop().match(containerReg) || [])[1]
+  if (containerId) data.containerId = containerId
+
+  const podId = (parts.pop().match(podReg) || [])[1]
+  if (podId) data.podId = podId.replace(/_/g, '-')
+
+  const taskId = (path.match(taskReg) || [])[1]
+  if (taskId) data.taskId = taskId
+
+  // if we reach the end and there's still no conatinerId match
+  // and there's not an ECS metadata file, try the ECS regular
+  // expression in order to get a container id in fargate
+  if (!containerId && !ecsMetadata) {
+    const basename = path.split('/').pop()
```

Comment:
Perhaps `.pop()` the `parts` array above into `basename` above so we don't need to `path.split('/')` twice.

Reply:
Sure, small enough change.  Done in 9f7521f5a98c3058f39f42924061a51aa15ea817. 

After a few years of working with you I realize your approach is very "let's gut what's there while we can" while I'm very "let's not touch that old fuse box because who knows what will catch on fire" :)